nexus/tests/integration_tests/unauthorized.rs

```diff
@@ -83,12 +101,121 @@ async fn test_unauthorized(cptestctx: &ControlPlaneTestContext) {
             .unwrap();
     }
 
-    // Verify the hardcoded endpoints.
+    // Create test data with non-hardcoded values.
```

Comment:
If I understand correctly: the changes in this file are intended to support creating objects that aren't so easily defined statically as the existing `SetupReq` pattern.  And the corresponding `VerifyEndpoint` things that use them also have to be dynamic?

This is kind of a bummer.  I was really hoping to preserve the more declarative approach because I'd like to use this metadata for other kinds of testing (e.g., stress-testing by running through the setup requests once and then hitting various "verifyendpoint" endpoints in random orders and with varying levels of concurrency).

So I'm curious why the objects we're creating here have to be created dynamically.  Is it just that they reference URLs that we don't know until we've set up the httptest server?  If that's the case I wonder if we should create the server in the `lazy_static` block where we create the setup requests so that we _can_ know it when we need it and then keep everything else working the same way it was.

Reply:
6597d670 creates the http server in lazy_static, then uses it, and reverts some of the changes I made. Thanks for the pointer, I didn't know lazy_static could have functions too.